src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -14,6 +14,9 @@ SET_SUBSYS(seastore_device);
 
 #define SECT_SHIFT	9
 #define RESERVED_ZONES 	1
+#define TAIL_BUF_SIZE	4096
+#define MAX_WRITE_BUF_SIZE 1048576
```

Comment:
This is used to select write buffer size to fill the zone from it's write pointer to the end of zone.
I observed many segments/zones getting closed with still around 2MB to 8MB empty space remaining, so for advance_wp(), 4KB writes would simply generate more write requests wasting the disk bandwidth. 
So as a compromise, only for the last 1MB or less of advancing the WP, I use 4KB writes. Writing 1MB at a time fills up the segment faster.
And also initally when crimson-osd comes up, it closes a lot of segments(around 15 to 16) after hardly writing any data(less than 1%) into the segment, so filling the rest of the 99% segment with 1MB writes are faster.  Since the amount of Lba's to advance, will not always be aligned to 1MB, the last part will be filled by 4K write requests.

It does not apply to ZNSSegment::write() because the caller provides the write buffer size and the data buffer.